agent/utils/ephemeral_ports.go

```diff
@@ -129,6 +158,14 @@ func GetHostPortRange(numberOfPorts int, protocol string) (string, error) {
 	}
 	if n != numberOfPorts {
 		return "", fmt.Errorf("%v contiguous host ports unavailable", numberOfPorts)
+	} else {
+		if resultEndPort == endHostPortRange {
+			// we've reached the end of the underlying ephemeral range, so circle back to the start
+			tracker.SetLastAssignedHostPort(startHostPortRange - 1)
+		} else {
+			// update tracker to point to the last assigned port
+			tracker.SetLastAssignedHostPort(resultEndPort)
+		}
 	}
-	return strconv.Itoa(startPort) + "-" + strconv.Itoa(lastPort), nil
+	return strconv.Itoa(resultStartPort) + "-" + strconv.Itoa(resultEndPort), nil
```

Comment:
Minor nit; `fmt.Sprintf()` could make this look a little nicer, but it's not a big deal.

Reply:
thanks, updated.